### ISSUE TITLE
Special characters in file name prevent downloading

### ISSUE THREAD

**Issue by lllama** [koltyakov/gosip#78]
If my file name contains some special characters (in my case, '#') then `GetFile` will fail when I use `ServerRelativeUrl`.

**Versions**
SharePoint Online

**To Reproduce**
Create a file with '#' in the name and attempt to download it by passing its `ServerRelativeUrl` to `GetFile`.

**Expected behavior**
The file to be downloaded. Instead we can a 404 error.

**Comment by lllama** [comment]
Ah - looks like `GetFileByPath` should be used to deal with this. Closing - sorry for the noise.

**Comment by lllama** [comment]
Looks like we still need to do some escaping ourselves, however.

**Comment by cody-lettau** [comment]
@lllama I'm looking to support both modern and classic UI. Currently we are using ```GetFile```, but we appear to be running into this issue. Looks like moving to ```GetFileByPath``` is not a viable solution since we need to support Classic UI as well.

Is the path forward for us to escape the ```ServerRelativeURL``` on our side before calling ```GetFile```?

**Comment by koltyakov** [comment]
It's not about classic and modern UI, but SharePoint version. If you need to support SharePoint 2013 or 2016 when you'd need to fallback to GetFile, otherwise with 2019 and Online no matter a UI choice you can use GetFileByPath.

**Comment by cody-lettau** [comment]
Okay, we have been using ```GetFile``` and plan to only support 2019 and later. So we should move to ```GetFileByPath```? I just see the comment [here](https://github.com/koltyakov/gosip/blob/46a1ea645a98e5a9fa4be67bbe1742125d4d17ff/api/web.go#L342) about only supported in modern SharePoint (guessing that's just meaning 2019+ and not the UI).

**Comment by cody-lettau** [comment]
Should we then be escaping the ```ServerRelativeURL``` on our side before making the call?

**Comment by koltyakov** [comment]
Yes, please. And also please report which special chars you'd got to escape, so I can extend the library underthehoods correspondingly.

**Comment by cody-lettau** [comment]
No luck with either approach to actually download the file content when uploaded via the modern UI (classic UI works just fine) in an on-prem 2019 SE environment. No error, just empty response being returned. Will keep digging as I'm not sure it's related to the SDK.
